sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -733,11 +734,16 @@ private <T> Mono<T> populateRequestConditionsAndContext(BlobRequestConditions re
 
                 requestConditionsFinal.setIfMatch(response.getValue().getETag());
                 Mono<T> result = downloadCall.get();
-                if (response.getValue().getMetadata().get(ENCRYPTION_DATA_KEY) != null) {
+
+                Map<String, String> caseInsensitiveMetadata = new HashMap<>();
+                for (Map.Entry<String, String> entry : response.getValue().getMetadata().entrySet()) {
+                    caseInsensitiveMetadata.put(entry.getKey().toLowerCase(Locale.ROOT), entry.getValue());
+                }
+                if (caseInsensitiveMetadata.get(ENCRYPTION_DATA_KEY) != null) {
```

Comment:
I think wee also need to convert `ENCRYPTION_DATA_KEY`'s value to lower case here.